Project1/PartyBase.h

```diff
@@ -29,7 +30,10 @@ class PartyBase {
 	iterator getPosition(const Unit& unit);
 
 	void insertUnit(UnitData unit);
-	void insertUnit(Unit& unit);
+	void insertUnit(UnitPtr& unit);
```

Comment:
So when you take unique_ptr by reference, it won't transfer ownership to you. The ptr is on the caller's stack, so it dies when he returns. You need to take the argument by value, which means they have to move it into your stack (since you can't copy unique_ptr), or take it by r-value reference `T&&` which makes them move it into your stack even if you _could_ copy them.